src/entities/communitySbt.ts

```diff
@@ -174,6 +205,166 @@ class SBT {
+        console.log(decoded[0])
+        const metadataURI = decoded[0].metadataURI;
+        const filenamme = metadataURI.split('/')[3];
+        const badgeType = parseInt(filenamme.split('.')[0]);
+
+        return badgeType;
+    }
+
+    public decodeMultipleBadgeTypes(input: Bytes): number[] {
+        let badgeTypes = new Array<number>;
+        const inter = new ethers.utils.Interface(CommunitySBT__factory.abi);
+        const decoded = inter.decodeFunctionData("multiRedeem", input);
+        for (const leafInfo of decoded[0]) {
+            console.log(leafInfo);
+            const metadataURI = leafInfo.metadataURI;
+            const filenamme = metadataURI.split('/')[3];
+            badgeTypes.push(parseInt(filenamme.split('.')[0]));
+        }
+        return badgeTypes;
+    }
+
+    async claimedBadgesInSubgraph(subgraphUrl: string, userAddress: string, season: number): Promise<Array<BadgeWithStatus>> {
+        const badgeQuery = `
+            query( $id: BigInt) {
+                badges(first: 50, where: {seasonUser_contains: $id}) {
+                    id
+                    badgeType
+                    badgeName
+                    awardedTimestamp
+                    mintedTimestamp
+                }
+            }
+        `;
+        const client = new ApolloClient({
+            cache: new InMemoryCache(),
+            link: new HttpLink({ uri: subgraphUrl, fetch })
+        })
+        const id = `${userAddress}#${season}`
+        const data = await client.query({
+            query: gql(badgeQuery),
+            variables: {
+                id: id,
+            },
+        });
+
+        let badgesClaimed = new Array<BadgeWithStatus>();
+        for (const badge of data.data.badges) {
+            badgesClaimed.push({
+                badgeType: badge.badgeType, 
+                claimingStatus: data.data.badge.mintedTimestamp == 0 ? 
+                    BadgeClaimingStatus.NOT_CLAIMED 
+                    : BadgeClaimingStatus.CLAIMED // only from subgraph's perspective
+            });
+        }
+
+        return badgesClaimed;
+    }
+
+    public getMethodId(isMultiRedeem: boolean, network: string): string {
+        switch (network) {
+            case "goerli":
+                return isMultiRedeem ? "0xbdb05092" : "0xbdb05092";
+            case "mainnet":
+                return isMultiRedeem ? "0x79a4aaa3" : "0x79a4aaa3";
+            default:
+                return "";
+        }
+    }
+
+    public getEtherscanURL(network: string, apiKey: string, userAddress: string): string {
+        switch (network) {
+            case "goerli":
+                return `https://api-goerli.etherscan.io/api?module=account&action=txlist&address=${userAddress}&page=1&offset=50&sort=desc&apikey=${apiKey}`
+            case "mainnet":
+                return `https://api-goerli.etherscan.io/api?module=account&action=txlist&address=${userAddress}&page=1&offset=50&sort=desc&apikey=${apiKey}`
```

Comment:
fix the url for mainnet, seems copy pasted

Reply:
fixed in d6bd73e951951d08b925ccfe8c6a7cb4c91861f6